lib/concurrent-ruby/concurrent/timer_task.rb

```diff
@@ -74,6 +85,12 @@ module Concurrent
   #
   #   #=> 'Boom!'
   #
+  # @example Configuring `:interval_type` with either :fixed_delay or :fixed_rate, default is :fixed_delay
+  #   task = Concurrent::TimerTask.new(execution_interval: 5, interval_type: :fixed_rate) do
```

Comment:
I personally consider the `interval_type` to be a static configuration of the behaviour of the application rather than something decided at run-time. Therefore, I wonder if this would be better implemented using sub-classes which implement the specific behaviour required, e.g. `FixedRateTimerTask`.

Reply:
Subclasses seem rather heavy for this, and I am concerned that for compatibility we would need to keep Concurrent::TimerTask so it would be both a leaf class and an abstract class, that sounds very confusing.
So I think a constructor argument like here makes most sense and is consistent with what is already done in concurrent-ruby APIs.

Reply:
Why would it be an abstract class?

Reply:
> Why would it be an abstract class?

If you make subclasses the superclass should be abstract. Otherwise I think it's pretty confusing.